testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1836,6 +1836,27 @@ public void testDoubleUpdateAndThenReadFromHive()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL)
+    public void testDeleteWithNoPredicateOnTableCreatedWithCtas()
+    {
+        try {
+            onTrino().executeQuery("SET SESSION scale_writers = true");
+            onTrino().executeQuery("SET SESSION writer_min_size = '4kB'");
+            onTrino().executeQuery("SET SESSION task_writer_count = 2");
+            withTemporaryTable("test_delete_with_no_predicate_on_ctas", true, false, NONE, tableName -> {
+                onTrino().executeQuery(
+                        "CREATE TABLE test_delete_with_no_predicate_on_ctas with (transactional = true) as select * from tpch.sf1000.orders limit 100000");
```

Comment:
`withTemporaryTable` allocates new table name. use `tableName` in queries instead of fixed name, to ensure cleanup.